types/mongoose/index.d.ts

```diff
@@ -1623,6 +1623,7 @@ declare module "mongoose" {
      * @param bool defaults to true
      */
     lean(bool?: boolean): Query<Object>;
+    lean<U>(bool?: boolean): Query<U>;
```

Comment:
See the bullet [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) starting with `getMeAT<T>(): T`; this is not the preferred usage of type parameters.

Reply:
@uniqueiniquity Thanks, updated.